dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,239 @@
+    //   const otherFavorites = prevFavorites.filter((r) => r.name !== repo.name);
+    //   if (isFavoriting) saveFavorites([...otherFavorites, repo]);
+    //   else saveFavorites(otherFavorites);
+    //   return isFavoriting ? [...otherFavorites, repo] : otherFavorites;
+    // }
+
+  const onNavToggle = () => {
+    setIsNavOpen(!isNavOpen);
+  };
+  const isRepoFavorited = (repo) => {
+    for (let i = 0; i < favoriteRepoNames.length; i++) {
+      if (
+        repo.name === favoriteRepoNames[i].name &&
+        repo.controller === favoriteRepoNames[i].controller &&
+        repo.metadata["dataset.created"] ===
+          favoriteRepoNames[i].metadata["dataset.created"]
+      )
+        return true;
+    }
+    return false;
+  };
+  const getSortableRowValues = (publicData) => {
+    const { controller, name } = publicData;
+    const creationDate = publicData.metadata["dataset.created"];
+    return [controller, name, creationDate];
+  };
+  let sortedRepositories = publicData;
+  if (activeSortIndex !== null) {
+    sortedRepositories = publicData.sort((a, b) => {
+      const aValue = getSortableRowValues(a)[activeSortIndex];
+      const bValue = getSortableRowValues(b)[activeSortIndex];
+      if (aValue === bValue) {
+        return 0;
+      }
+      if (activeSortDirection === "asc") {
+        return aValue > bValue ? 1 : -1;
+      } else {
+        return bValue > aValue ? 1 : -1;
+      }
+    });
+  }
+  const getSortParams = (columnIndex) => ({
+    isFavorites: columnIndex === 3,
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortIndex(index);
+      setActiveSortDirection(direction);
+    },
+    columnIndex,
+  });
+  const handleButtonClick = (_isSelected, event) => {
+    const id = event.currentTarget.id;
+    setIsSelected(id);
+  };
+  const setControllerName = (controllerNameValue) => {
+    controllerName = controllerNameValue;
+  };
+  const setDateRange = (startDateValue, endDateValue) => {
+    startDate = startDateValue;
+    endDate = endDateValue;
+  };
+  const saveFavorites = (fav) => {
+    localStorage.setItem("favControllers", JSON.stringify(fav));
+  };
+  const Sidebar = (
+    <PageSidebar
+      nav={<NavItems />}
+      className="sidebar"
+      isNavOpen={isNavOpen}
+    />
+  );
+  const NavbarDrawer = () => {
+    return (
+      <Masthead id="basic">
+        <MastheadToggle>
+          <PageToggleButton isNavOpen={isNavOpen} onNavToggle={onNavToggle}>
+            <BarsIcon />
+          </PageToggleButton>
+        </MastheadToggle>
+      </Masthead>
+    );
+  };
+  const selectedArray =
+    isSelected === "firstButton" ? publicData : favoriteRepoNames;
+  return (
+    <>
+      <Page header={<NavbarDrawer />} sidebar={Sidebar}>
+        <AlertMessage
+          message="Want to see only metric relevant to you?"
+          link="Login to create an account"
```

Comment:
Yes,I just blindly copied the line from the wireframe,will fix it.

Reply:
Actually, since a user has to already have an account before they can upload data, the concept here should be, "Want to see your own data?  Log into your account".  If they don't have an account, we can catch that on the login page when they get there.

Reply:
I have changed the Alert message link text from "Click here to create an account" to "Log into your account"

Reply:
@shubham-html-css-js, I don't see the change that you're referring to...should it show up in the code here, or is it somewhere else?

Reply:
I am not sure why it is not being showed.Can you check [here](https://github.com/distributed-system-analysis/pbench/pull/2733/commits/9fa4d03b654d2ff8680a22e86b868d13e33d6672)

Reply:
OK, the problem is in me.  😀

I do see the change that you made to `dashboard/src/modules/components/AlertComponent/index.jsx`; but how does that interact with the `<AlertMessage>` tag, here in `dashboard/src/modules/components/TableComponent/index.jsx`?

Reply:
We are calling the `<AlertMessage>` component in `dashboard/src/modules/components/TableComponent/index.jsx` passing the helper message and the link text as props to the `<AlertMessage>` component ,who on recieving the props display them wrapping them with appropriate tags (For example,link text is wrapped with <a> tag).

Reply:
OK, I'm starting to understand.  🙂

So, the code here passes the `message` which the code there displays...but the code here is also passing a `link` (text) which code there is ignoring in favor of using `Login to create an account` instead.

Should the code there be using the value of `link`, or should we not bother specifying it here, or what?

Reply:
Ah, nevermind, you just fixed it.  🙂